File: GymCRM.SchedulingAPI/Models/DTOs/TrainerDailyAvailability.cs
using GymCRM.SchedulingAPI.Models.DTOs;

namespace GymCRM.SchedulingAPI.Models.DTOs;

public class TrainerDailyAvailability
{
    public Guid Id { get; set; }
    public string DayOfWeek { get; set; }
    public bool IsDayOff { get; set; }
    public DateTime DateCreatedUtc { get; set; }
    public DateTime DateModifiedUtc { get; set; }
    public List<TrainerWorkingHours> WorkingHours { get; set; }
}

File: gymcrm.web/src/modules/scheduling/types/trainerDailyAvailability.ts
import {TrainerWorkingHours} from "./trainerWorkingHours";

// Mirrors GymCRM.SchedulingAPI.Models.DTOs.TrainerDailyAvailability 1:1.
// dayOfWeek must be one of System.DayOfWeek's names (see constants/daysOfWeek.ts).
export interface TrainerDailyAvailability {
    id: string;
    dayOfWeek: string;
    isDayOff: boolean;
    dateCreatedUtc: string;
    dateModifiedUtc: string;
    workingHours: TrainerWorkingHours[];
}
